src/app/modules/zxing-scanner/browser-code-reader.ts

```diff
@@ -137,6 +156,27 @@ export class BrowserCodeReader {
         this.videoElement.addEventListener('loadedmetadata', this.videoLoadedMetadataEventListener);
     }
 
+    private checkTorchCompatibility(): void {
+        this.track = this.stream.getVideoTracks()[0];
+
+        const imageCapture = new ImageCapture(this.track);
+        const photoCapabilities = imageCapture.getPhotoCapabilities().then((capabilities) => {
+            this.torchCompatible = !!capabilities.torch;
+        });
+    }
+
+    public setTorch(on: boolean): void {
+        if (this.torchCompatible) {
+            this.track.applyConstraints({
+                advanced: [<any>{ torch: on }]
+            });
+        }
+    }
+
+    public get torchAvailable() {
+        return this.torchCompatible;
```

Comment:
Here we have a syncronous thing depending on a assyncronously setted variable value. Maybe making it all async should make things safier.